src/main/java/org/gridsuite/filter/server/FilterService.java

```diff
@@ -476,6 +476,11 @@ private List<Identifiable<?>> get2WTransformerList(Network network, AbstractFilt
             return network.getTwoWindingsTransformerStream()
                     .filter(twoWindingsTransformer -> equipmentIds.contains(twoWindingsTransformer.getId()))
                     .collect(Collectors.toList());
+        } else if (filter instanceof ExpertFilter expertFilter) {
+            var rule = expertFilter.getRules();
+            return network.getTwoWindingsTransformerStream()
+                .filter(rule::evaluateRule)
+                .collect(Collectors.toList());
```

Comment:
je pense que tu peux utiliser .toList()

Reply:
`.toList()` returns a list of `TwoWindingsTransformer` but we need to return a list of `Identifiable<?>`
I prefer keeping it like this rather than casting the returned list